lib/daru/accessors/array_wrapper.rb

```diff
@@ -67,26 +67,24 @@ def dup
       end
 
       def mean
-        sum.quo(@size - @context.missing_positions.size).to_f
+        return nil if @data.compact.empty?
```

Comment:
You'll call `@data.compact` twice in case its not empty, which will be expensive. To top it all your code does not consider the fact that elements apart from `nil` can also be marked as missing.